File: content/code-examples/tested/csharp/driver/TimeSeries/SecondaryIndexes.snippet.sort-and-group.cs
var pipeline = new EmptyPipelineDefinition<BsonDocument>()
    .Sort(new BsonDocument
    {
        { "metadata.sensorId", 1 },
        { "timestamp", -1 }
    })
    .Group(new BsonDocument
    {
        { "_id", "$metadata.sensorId" },
        { "ts", new BsonDocument("$first", "$timestamp") },
        { "temperatureF", new BsonDocument("$first", "$currentConditions.tempF") }
    });

var result = _collection?.Aggregate(pipeline).ToList();

File: content/code-examples/tested/javascript/driver/time-series/secondary-indexes/secondary-indexes.snippet.sort-and-group.js
const pipeline = [
  { $sort: { 'metadata.sensorId': 1, timestamp: -1 } },
  {
    $group: {
      _id: '$metadata.sensorId',
      ts: { $first: '$timestamp' },
      temperatureF: { $first: '$currentConditions.tempF' },
    },
  },
];

const result = await collection.aggregate(pipeline).toArray();
